### PR TITLE
Stream debug

### PR DESCRIPTION
**Pull request by kgopal492**
Hashtag functionality basically works. Still need to add tests and edge case checks.

### CODE REVIEW

File: cpp/service.cc [hunk 1]
```diff
@@ -1,13 +1,15 @@
 #include "service.h"
 
 ServiceLayer::ServiceLayer(KeyValueClientInterface* key_value_connection) {
+  service_mtx_ = new std::mutex();
```
Comment (barath) [inline cpp/service.cc:4]: Generally you want to make a mutex a non-pointer member variable and then just initialize it in the initializer list.

File: cpp/service.cc [hunk 2]
```diff
@@ -120,17 +124,21 @@ void ServiceLayer::storeHashtags(const std::string& text, const std::string& chi
...
     if (hashtag.length() != 0) {
-      const std::string hashtag_key = khashtag_ + hashtag;
-      const std::deque<std::string>& hashtag_chirp_ids = store_->get(hashtag_key);
+      std::deque<std::string> hashtag_streamers = store_->get(khashtagStreamers_ + hashtag);
       bool chirp_added = false;
-      for (std::string curr_hash_chirp_id : hashtag_chirp_ids) {
-        if(curr_hash_chirp_id == chirp_id) {
-          chirp_added = true;
-          break;
+      for (std::string streamer : hashtag_streamers) {
```
Comment (barath) [inline cpp/service.cc:129]: Minor: const ref.

**Review by barath — APPROVED** [review]

**Review by jillianKhoo — APPROVED** [review]
